### ISSUE TITLE
CSS Grid - Sequence of Breakpoints

### ISSUE THREAD

**Issue by gisu** [postcss/autoprefixer#1085]
Another error. I process a grid using two breakpoints, that is, three grid templates. 

The order is No Breakpoint > from 768px > 960px. 

If Autoprefixer now creates the fallbacks for Internet Explorer, the order of the breakpoints does not match. 

```scss
.c-footerNav {
  display: grid;
  grid-gap: 30px 0;
  grid-template:
    "fnbox1" auto
    "fnbox2" auto
    "fnbox3" auto
    "fnbox4" auto /
     1fr;

  @include break('m') {
    grid-gap: 40px;
    grid-template:
      "fnbox1 fnbox1 fnbox1" auto
      "fnbox2 fnbox3 fnbox4" auto /
       1fr  1fr  1fr
  }

  @include break('ml') {
    grid-gap: 60px 40px;
    grid-template:
      "fnbox1 fnbox2 fnbox3 fnbox4" auto /
       1fr  1fr  1fr  1fr;
  }
  &__box1 {
    grid-area: fnbox1;
  }
  &__box2 {
    grid-area: fnbox2;
  }
  &__box3 {
    grid-area: fnbox3;
  }
  &__box4 {
    grid-area: fnbox4;
  }
}
```

Output

```css
.c-footerNav {
  display: -ms-grid;
  display: grid;
  grid-gap: 30px 0;
  -ms-grid-rows: auto 30px auto 30px auto 30px auto;
  -ms-grid-columns: 1fr;
  grid-template: "fnbox1" auto "fnbox2" auto "fnbox3" auto "fnbox4" auto / 1fr;
}

@media screen and (min-width: 768px) {
  .c-footerNav {
    grid-gap: 40px;
    -ms-grid-rows: auto 40px auto;
    -ms-grid-columns: 1fr 40px 1fr 40px 1fr;
    grid-template: "fnbox1 fnbox1 fnbox1" auto "fnbox2 fnbox3 fnbox4" auto / 1fr 1fr 1fr;
  }
}

@media screen and (min-width: 960px) {
  .c-footerNav {
    grid-gap: 60px 40px;
    -ms-grid-rows: auto;
    -ms-grid-columns: 1fr 40px 1fr 40px 1fr 40px 1fr;
    grid-template: "fnbox1 fnbox2 fnbox3 fnbox4" auto / 1fr 1fr 1fr 1fr;
  }
}

.c-footerNav__box1 {
  -ms-grid-row: 1;
  -ms-grid-column: 1;
  grid-area: fnbox1;
}

.c-footerNav__box2 {
  -ms-grid-row: 3;
  -ms-grid-column: 1;
  grid-area: fnbox2;
}

.c-footerNav__box3 {
  -ms-grid-row: 5;
  -ms-grid-column: 1;
  grid-area: fnbox3;
}

.c-footerNav__box4 {
  -ms-grid-row: 7;
  -ms-grid-column: 1;
  grid-area: fnbox4;
}

@media screen and (min-width: 960px) {
  .c-footerNav__box1 {
    -ms-grid-row: 1;
    -ms-grid-column: 1;
  }
  .c-footerNav__box2 {
    -ms-grid-row: 1;
    -ms-grid-column: 3;
  }
  .c-footerNav__box3 {
    -ms-grid-row: 1;
    -ms-grid-column: 5;
  }
  .c-footerNav__box4 {
    -ms-grid-row: 1;
    -ms-grid-column: 7;
  }
}

@media screen and (min-width: 768px) {
  .c-footerNav__box1 {
    -ms-grid-row: 1;
    -ms-grid-column: 1;
    -ms-grid-column-span: 5;
  }
  .c-footerNav__box2 {
    -ms-grid-row: 3;
    -ms-grid-column: 1;
  }
  .c-footerNav__box3 {
    -ms-grid-row: 3;
    -ms-grid-column: 3;
  }
  .c-footerNav__box4 {
    -ms-grid-row: 3;
    -ms-grid-column: 5;
  }
}
```

What else struck me, the grid area names don't seem to be scoped. If I use "box1" several times as grid area name in various grids, the autoprefixer will no longer create fallbacks for subsequent grids.

**Comment by Dan503** [comment]
That is not a bug, that's a feature.

See issue https://github.com/postcss/autoprefixer/issues/1048 for the reason why.

**Comment by Dan503** [comment]
And as for the non-scoped area names, that is a restriction that can't really be avoided.

See issue https://github.com/postcss/autoprefixer/issues/1038 for details.

**Comment by Dan503** [comment]
Make sure to read this article before attempting to use the Autoprefixer CSS Grid translation:

https://css-tricks.com/css-grid-in-ie-css-grid-and-the-new-autoprefixer/

There are many restrictions that you need to live with if you want to use the IE implementation and that article explains them all very clearly.

**Comment by Dan503** [comment]
> That is not a bug, that's a feature.

Oh actually no, I take that back. It is a bug. They should stay in the same order that they are declared in.

**Comment by gisu** [comment]
The media queries are also my main problem, everything else can be avoided if you know it. 

I changed the MediaQueries of the grid templates `no Break > 960px > 768px` (which makes no sense but is just for testing). The order of the fallbacks was `no Break > 768px > 960px`. So always in the reverse order. This does not make handling easy. The only way I can process a maximum of one breakpoint.

**Comment by ai** [comment]
I will try to think for solution free days later. Resting after the conference rush.

For quick solution, try to change order of current rules.

**Comment by gisu** [comment]
I have already changed the order, the autoprefixer does not take the order of the breakpoints (where I adjust the grid template layout), but arranges them in reverse order for the fallbacks.

Grid Template Breakpoints : no Break > 768px > 960px
Autoprefixer fallback breakpoints: no Break > 960px > 768px

After changeover
Grid Template Breakpoints : no Break > 960px > 768px
Autoprefixer fallback breakpoints: no Break > 768px > 960px

Otherwise no stress, I just test what works :)

**Comment by ai** [comment]
@bogdan0083 maybe you want to fix it as well?

**Comment by bogdan0083** [comment]
@ai I think I'll try. I've already reproduced the bug and found the piece of code that causes it

**Comment by bogdan0083** [comment]
fixed here: #1087

**Comment by ai** [comment]
Released in 9.1.1
